controllers:
  l_ank_roll_controller:
    type: JointPositionController
    joint_name: l_ank_roll
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 1
      init: 500
      min: 0
      max: 1000
  l_ank_pitch_controller:
    type: JointPositionController
    joint_name: l_ank_pitch
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 3
      init: 500
      min: 0
      max: 1000
  l_knee_controller:
    type: JointPositionController
    joint_name: l_knee
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 5
      init: 240
      min: 0
      max: 1000
  l_hip_pitch_controller:
    type: JointPositionController
    joint_name: l_hip_pitch
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 7
      init: 500
      min: 0
      max: 1000
  l_hip_roll_controller:
    type: JointPositionController
    joint_name: l_hip_roll
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 9
      init: 500
      min: 0 
      max: 1000
  l_hip_yaw_controller:
    type: JointPositionController
    joint_name: l_hip_yaw
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 11
      init: 500
      min: 0
      max: 1000
  l_sho_pitch_controller:
    type: JointPositionController
    joint_name: l_sho_pitch
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 13
      init: 875
      min: 1000
      max: 0
  l_sho_roll_controller:
    type: JointPositionController
    joint_name: l_sho_roll
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 15
      init: 500
      min: 0
      max: 1000
  l_el_pitch_controller:
    type: JointPositionController
    joint_name: l_el_pitch
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 17
      init: 500
      min: 0
      max: 1000
  l_el_yaw_controller:
    type: JointPositionController
    joint_name: l_el_yaw
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 19
      init: 500
      min: 0
      max: 1000
  l_gripper_controller:
    type: JointPositionController
    joint_name: l_gripper
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 21
      init: 500
      min: 0
      max: 1000
  r_ank_roll_controller:
    type: JointPositionController
    joint_name: r_ank_roll
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 2
      init: 500
      min: 0
      max: 1000
  r_ank_pitch_controller:
    type: JointPositionController
    joint_name: r_ank_pitch
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 4
      init: 500
      min: 0
      max: 1000
  r_knee_controller:
    type: JointPositionController
    joint_name: r_knee
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 6
      init: 760
      min: 0
      max: 1000
  r_hip_pitch_controller:
    type: JointPositionController
    joint_name: r_hip_pitch
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 8
      init: 500
      min: 0
      max: 1000
  r_hip_roll_controller:
    type: JointPositionController
    joint_name: r_hip_roll
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 10
      init: 500
      min: 0
      max: 1000
  r_hip_yaw_controller:
    type: JointPositionController
    joint_name: r_hip_yaw
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 12
      init: 500
      min: 0
      max: 1000
  r_sho_pitch_controller:
    type: JointPositionController
    joint_name: r_sho_pitch
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 14
      init: 125
      min: 1000
      max: 0
  r_sho_roll_controller:
    type: JointPositionController
    joint_name: r_sho_roll
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 16
      init: 500
      min: 0
      max: 1000
  r_el_pitch_controller:
    type: JointPositionController
    joint_name: r_el_pitch
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 18
      init: 500
      min: 0
      max: 1000
  r_el_yaw_controller:
    type: JointPositionController
    joint_name: r_el_yaw
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 20
      init: 500
      min: 0
      max: 1000
  r_gripper_controller:
    type: JointPositionController
    joint_name: r_gripper
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 22
      init: 500
      min: 0
      max: 1000
  head_pan_controller:
    type: JointPositionController
    joint_name: head_pan
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 23
      init: 500
      min: 0
      max: 1000
  head_tilt_controller:
    type: JointPositionController
    joint_name: head_tilt
    joint_speed: 1.0
    port_id: 1
    servo:
      id: 24
      init: 500
      min: 0
      max: 1000

  ainex_controller:
    type: JointTrajectoryActionController
    joint_trajectory_action_node:
      min_velocity: 0.1
      constraints:
        goal_time: 0.05
    joint_controllers:
      - "l_ank_roll_controller"
      - "l_ank_pitch_controller"
      - "l_knee_controller"
      - "l_hip_pitch_controller"
      - "l_hip_roll_controller"
      - "l_hip_yaw_controller"
      - "l_sho_pitch_controller"
      - "l_sho_roll_controller"
      - "l_el_pitch_controller"
      - "l_el_yaw_controller"
      - "l_gripper_controller"
      - "r_ank_roll_controller"
      - "r_ank_pitch_controller"
      - "r_knee_controller"
      - "r_hip_pitch_controller"
      - "r_hip_roll_controller"
      - "r_hip_yaw_controller"
      - "r_sho_pitch_controller"
      - "r_sho_roll_controller"
      - "r_el_pitch_controller"
      - "r_el_yaw_controller"
      - "r_gripper_controller"
      - "head_pan_controller"
      - "head_tilt_controller"

serial_ports:
  - port_name: "/dev/ttyAMA0"
    port_id: 1
    baud_rate: 115200
    min_motor_id: 1
    max_motor_id: 30
    fake_read: true
    connected_ids: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 12, 22, 23, 24]
    update_rate: 40
